tasker/core/state_manager.py

```
@@ -94,6 +94,46 @@ def get_all_task_results(self) -> Dict[int, Dict[str, Any]]:
+        This prevents blocking other threads during potentially large file reads.
+
+        Args:
+            task_id: Task identifier
+            output_type: 'stdout' or 'stderr'
+
+        Returns:
+            Full output content, reading from temp file if available
+        """
+        # CRITICAL: Hold lock ONLY for copying values (fast operation)
+        with self._lock:
+            result = self._task_results.get(task_id, {})
```

Ruff (0.14.3)

Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)